source/simulator/assemblers/advection.cc

```diff
@@ -437,6 +437,11 @@ namespace aspect
                 }
             }
         }
+      else if (cell->has_periodic_neighbor (face_no))
```

Comment:
will this lead to assembling the term twice, once from each side of the periodic boundary?